src/app/hostname-generators/data/HostnameGenerator.ts

```diff
@@ -0,0 +1,71 @@
+import type { components, paths } from '@/types/auto-generated'
+
+// TODO(schogges): Remove `creationTime` and `modificationTime` once the api specs are complete
+export type HostnameGeneratorList = Omit<components['responses']['HostnameGeneratorList']['content']['application/json'], 'items'> & {
+  items: NonNullable<components['responses']['HostnameGeneratorList']['content']['application/json']['items']>[number] & {
+    creationTime: string
+    modificationTime: string
+  } | undefined
+}
+export type HostnameGeneratorItem = components['responses']['HostnameGeneratorItem']['content']['application/json'] & {
+  creationTime: string
+  modificationTime: string
+}
+export type HostnameGeneratorGetParams = paths['/hostnamegenerators/{name}']['get']['parameters']
+
+export interface HostnameGenerator extends HostnameGeneratorItem {
+  id: string
+  namespace: string
+  zone: string
+  mesh: string
+  selector?: {
+    routeName: string
+    label: string
+  } | undefined
+  $raw: HostnameGeneratorItem
+}
+
+export const HostnameGenerator = {
+  fromObject(item: HostnameGeneratorItem) {
+    const labels = item.labels ?? {}
+    const name = labels['kuma.io/display-name'] ?? item.name
+    const namespace = labels['k8s.kuma.io/namespace'] ?? ''
+    return {
+      ...item,
+      id: item.name,
+      name,
+      namespace,
+      zone: labels['kuma.io/origin'] === 'zone' && labels['kuma.io/zone'] ? labels['kuma.io/zone'] : '',
+      mesh: labels['kuma.io/mesh'] || 'default',
```

Comment:
I am also not too sure about that, but the main driver for this decision was because I added a link to the mesh service on the details page. Iirc somewhere in the docs that `default` is the default for a `mesh`. Another thing is that if I am not mistaken, the hostname generators are only for `MeshService`, `MeshExternalService` and `MeshMultiZoneService`, which means that there is always a mesh, right?

Reply:
As far as I understood, HostnameGenerators are a top-level thing, i.e. they don't belong to a mesh they span the entire install/cluster.

> I am also not too sure about that, but the main driver for this decision was because I added a link to the mesh service on the details page (mesh is a param for the URL).

Oh! Glad you pointed this out 👍  not totally sure if we can link into a specific mesh from a HostnameGenerator 🤔

Reply:
The selector `meshService`, `meshExternalService` or `meshMultiZoneService` together with a `mesh` would allow to link to the respective service type view 🤔

Reply:
We chatted about this, and saw that we can't use the top-level `kuma.io/mesh` for linking, if anything we'd generate any links form `matchLabels` instead. If we decide to do that this will be a future PR. We also removed the defaulting to `default` for `kuma.io/mesh`. In the future, from what I understand this top-level label will be removed entirely from HGs.